tests/api2/test_530_ups.py

```diff
@@ -28,25 +36,92 @@
     'monpwd'
 ]
 
+default_dummy_data = {
+    'battery.charge': 100,
+    'driver.parameter.pollinterval': 2,
+    'input.frequency': 49.9,
+    'input.frequency.nominal': 50.0,
+    'input.voltage': 230,
+    'input.voltage.nominal': 240,
+    'ups.status': 'OL',
+    'ups.timer.shutdown': -1,
+    'ups.timer.start': -1,
+}
+
+
+def remove_file(filepath):
+    ssh(f'rm {filepath}', check=False)
+
+
+def did_shutdown():
+    return ssh(f'cat {SHUTDOWN_MARKER_FILE}', check=False) == "done\n"
+
+
+def write_fake_data(data={}):
+    all_data = default_dummy_data | data
+    with NamedTemporaryFile() as f:
+        for k, v in all_data.items():
+            f.write(f'{k}: {v}\n'.encode('utf-8'))
+        f.flush()
+        os.fchmod(f.fileno(), 0o644)
+        results = send_file(f.name, DUMMY_FAKEDATA_DEV, user, password, truenas_server.ip)
+        assert results['result'], str(results['output'])
+
+
+def wait_for_alert(klass, retries=10):
+    assert retries > 0
+    while retries:
+        alerts = call('alert.list')
+        for alert in alerts:
+            if alert['klass'] == klass:
+                return alert
+
+
+@pytest.fixture(scope='module')
+def ups_running():
+    with ensure_service_enabled('ups'):
+        with ensure_service_started('ups'):
+            yield
+
+
+@pytest.fixture(scope='module')
+def dummy_ups_driver_configured():
+    write_fake_data()
+    remove_file(SHUTDOWN_MARKER_FILE)
+    old_config = call('ups.config')
+    del old_config['complete_identifier']
+    del old_config['id']
+    payload = {
+        'mode': 'MASTER',
+        'driver': 'dummy-ups',
+        'port': DUMMY_FAKEDATA_DEV,
+        'description': 'dummy-ups in dummy-once mode',
+        'shutdowncmd': f'echo done > {SHUTDOWN_MARKER_FILE}'
+    }
+    call('ups.update', payload)
+    try:
+        yield
+    finally:
+        call('ups.update', old_config)
+        remove_file(SHUTDOWN_MARKER_FILE)
+        remove_file(DUMMY_FAKEDATA_DEV)
+
 
 def test_01_get_ups_service_id():
```

Comment:
I know you didn't write this test but we need to improve it while you're here. `global` is a keyword that should be avoided at all costs, even in tests. For example, this function just literally queries the service by name and gets the database id. However, the only other functions that use thisid is in `test_26` and `test_28`. There is no reason for that. We can simply query by service name.

In another test file I added a function that queries the service state instead of writing `service.query [["...` a bunch of times.
```
def get_service_state():
    return call(service.query, [['service', '=', 'ups']], {'get': True})
```